scripts/run_e2e_tests.py

```diff
@@ -641,71 +565,34 @@ def run_tests(args=None):
-                ciphertext = enc_file.read()
-                plaintext = simplecrypt.decrypt(
-                    google_auth_decode_password, ciphertext).decode('utf-8')
-                dec_file.write(plaintext)
-
-        sheets_scopes = ['https://www.googleapis.com/auth/spreadsheets']
-        creds = service_account.Credentials.from_service_account_file(
-            'auth.json', scopes=sheets_scopes)
-        sheet = googleapiclient.discovery.build(
-            'sheets', 'v4', credentials=creds).spreadsheets()
-        flaky_tests_list = get_flaky_tests_data_from_sheets(sheet)
-
-    suite_name = parsed_args.suite.lower()
-    if len(flaky_tests_list) > 0 and p.returncode != 0:
-        for i, line in enumerate(output_lines):
-            if line == '*                    Failures                    *':
-                test_name = output_lines[i + 3][3:].strip().lower()
-
-                # Remove coloring characters.
-                ansi_escape = re.compile(
-                    r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
-                failure_log = ansi_escape.sub('', output_lines[i + 4])
-                failure_log = failure_log.strip().lower()
-                for index, row in enumerate(flaky_tests_list):
-                    flaky_suite_name = row[0].strip().lower()
-                    flaky_test_message = row[1].strip().lower()
-                    flaky_error_message = row[2].strip().lower()
-                    if (
-                            suite_name == flaky_suite_name or
-                            flaky_suite_name == '[general]'):
-                        if (
-                                test_name == flaky_test_message or
-                                flaky_test_message == 'many'):
-                            if flaky_error_message in failure_log:
-                                update_flaky_tests_count(sheet, index, row[3])
-                                try:
-                                    cleanup_portserver(portserver_process)
-                                    cleanup()
-                                except Exception: # pragma: no cover
-                                    # This is marked as no cover because the
-                                    # exception happens due to some processes
-                                    # running on the local system, which might
-                                    # interfere with the cleanup stuff. This is
-                                    # added as a failsafe to make sure that
-                                    # even when it throws an exception, the
-                                    # test is retried.
-                                    pass # pragma: no cover
-                                return 'flake'
-    sys.exit(p.returncode)
+        try:
+            output_lines.append(python_utils.UNICODE(nextline.strip()))
+        except UnicodeDecodeError:
+            pass
```

Comment:
What's an example of a normal line that would yield a `UnicodeDecodeError`? (i.e. why not raise an error here)

Reply:
I'm not sure, it was added by Nitesh as we had some Unicode error issue with some text which comparing.

Reply:
Where did Nithesh add the `UnicodeDecodeError` catching code? I don't see it anywhere in the codebase ([search](https://github.com/oppia/oppia/search?q=UnicodeDecodeError)).